packages/frontend/src/components/common/Table/ScrollableTable/TableBody.tsx

```diff
@@ -79,34 +81,89 @@ const TableRow: FC<TableRowProps> = ({
                     getConditionalRuleLabel,
                 );
 
+                const toggleExpander = row.getToggleExpandedHandler();
+                const fontColor =
+                    conditionalFormattingColor &&
+                    readableColor(conditionalFormattingColor) === 'white'
+                        ? 'white'
+                        : undefined;
+
                 return (
                     <BodyCell
                         minimal={minimal}
                         key={cell.id}
                         style={meta?.style}
                         backgroundColor={conditionalFormattingColor}
-                        fontColor={
-                            conditionalFormattingColor &&
-                            readableColor(conditionalFormattingColor) ===
-                                'white'
-                                ? 'white'
-                                : undefined
-                        }
+                        fontColor={fontColor}
                         className={meta?.className}
                         index={index}
                         cell={cell}
                         isNumericItem={isNumericItem(meta?.item)}
                         hasData={!!meta?.item}
                         cellContextMenu={cellContextMenu}
                         isLargeText={
-                            (cellValue?.value.formatted || '').length >
+                            (cellValue?.value?.formatted || '').length >
                             SMALL_TEXT_LENGTH
                         }
                         tooltipContent={tooltipContent}
                     >
-                        {flexRender(
-                            cell.column.columnDef.cell,
-                            cell.getContext(),
+                        {cell.getIsGrouped() ? (
+                            <>
+                                <button
```

Comment:
this can be a mantine's `ActionIcon` component with minimal styling (https://v6.mantine.dev/core/action-icon/)

Reply:
Tried ActionIcon, but was not rendering nicely with both icon and count label being clickable. Tried UnstyledButton, but settled on Button with variant 'subtle`, but please let me know if we should change it.